### PR TITLE
Upgraded module to SS4 compatible version

### PR DESCRIPTION
**Pull request by Makreig**
This is an initial update for SS4 compatability.
There are still some minor improvements that need to be made, but this is fully working for me.

### CODE REVIEW

**Comment by Makreig** [comment]
almost forgot - i still need to work on the tests - they're not currently working.

**Comment by Makreig** [comment]
any chance you would be able to take a look at this @fspringveldt?

**Comment by fspringveldt** [comment]
Thanks for this @Makreig . I'll have a run through it and provide you with feedback shortly.

**Comment by Makreig** [comment]
I've now updated the unit tests to match my changes.

File: src/Control/UserController.php [hunk 1]
```diff
@@ -10,26 +34,25 @@ class UserController extends Controller implements PermissionProvider
...
+            ),
         );
     }
 
-
     public function index()
     {
         if (!Member::currentUserID()) {
-            return $this->redirect('/Security/login');
+            return $this->redirect('Security/login');
```
Comment (fspringveldt) [inline src/Control/UserController.php:53]: Mind doing below here too?
```
if(!Member::currentUserID()) {
$security = Injector::inst()->get(Security::class);
$this->assertContains("<a href=\"{$baseURL}//Security/login?BackURL=/\">", $body); 
$link = $link $security->Link('login');
return $this->redirect($link);
}
```

Comment (fspringveldt) [reply src/Control/UserController.php:53]: Nice work @Makreig. 

I'll accept as-is and do the minor updates myself.

Thanks.

**Comment by fspringveldt** [comment]
Nice work @Makreig. I'll do a few more minor adds and perform a release soon. Thanks.

**Pull request merged**